Sources/Logto/LogtoUtilities.swift

```diff
@@ -29,20 +30,65 @@ public enum LogtoUtilities {
     /// Decode ID Token claims WITHOUT validation.
     /// - Parameter token: The JWT to decode
     /// - Returns: A set of ID Token claims
-    static func decodeIdToken(_ token: String) throws -> IdTokenClaims {
+    static func decodeIdToken(_ idToken: String) throws -> IdTokenClaims {
         let decoder = JSONDecoder()
         decoder.keyDecodingStrategy = .convertFromSnakeCase
 
-        let segments = token.split(separator: ".")
+        let segments = idToken.split(separator: ".")
 
         guard let payload = segments[safe: 1] else {
-            throw LogtoErrors.Decode.noPayloadFound
+            throw LogtoErrors.Decoding.noPayloadFound
         }
 
         guard let decoded = String.fromUrlSafeBase64(string: String(payload)) else {
-            throw LogtoErrors.Decode.invalidUrlSafeBase64Encoding
+            throw LogtoErrors.Decoding.invalidUrlSafeBase64Encoding
         }
 
         return try decoder.decode(IdTokenClaims.self, from: Data(decoded.utf8))
     }
+
+    /// Verify the give ID Token:
+    /// * One of the JWKs matches the token.
+    /// * Issuer matches token payload `iss`.
+    /// * Client ID matches token payload `aud`.
+    /// * The token is not expired.
+    /// * The token is issued in +/- 1min.
+    static func verifyIdToken(
+        _ idToken: String,
+        issuer: String,
+        clientId: String,
+        publicKeys: [RSAPublicKey],
+        forTimeInterval: TimeInterval = Date().timeIntervalSince1970
+    ) throws {
+        if publicKeys.isEmpty {
+            throw LogtoErrors.Verification.missingJwk
+        }
+
+        // Public key verification
+        let jws = try JWS(compactSerialization: idToken)
```

Comment:
curious what exception would JWS lib throw if `idToken` decode failed here? Compared with the following `PublicKey` verification, should we all throw a general JWS verification failed exception?

Reply:
for now it'll throw `JOSESwiftError` or `DecodingError` I think it's good to go. like other repos, will it be better to throw "raw" errors when it's possible for easier debugging?